lib/instrumentation/http.js

```diff
@@ -20,47 +20,85 @@ let instrumentHTTP = (http, opts = {}) => {
     // node 10+ supports a three arg form.  Instead of writing both implementations,
     // dynamically switch on arg types.  I _hate_ code that does this, but it's easier
     // and cleaner than the alternative.
-    return function (_url, options, cb) {
+    //
+    // The http.request function can be called either as:
+    // http.request(options[, callback])
+    // or
+    // http.request(url[, options][, callback])
+    // So we can have 1-3 args to work with. For more details, see:
+    // http://nodejs.org/api/http.html#http_http_request_url_options_callback
+    return function (..._args) {
       if (!api.traceActive()) {
-        return original.apply(this, Array.from(arguments));
+        return original.apply(this, Array.from(_args));
```

Comment:
We can drop the `Array.from` now because variadic args are proper arrays already.
```javascript
function arrrrr(...args) {
    console.log("arguments is an array? ", Array.isArray(arguments))
    console.log("_args is an array? ", Array.isArray(_args))
}
foo()
// arguments is an array?  false
// _args is an array?  true
```